src/main/main.ts

```diff
@@ -12,6 +12,17 @@ import { SettingType, userSettings } from './config/settings';
 let jupyterApp: JupyterApplication;
 let fileToOpenInMainInstance = '';
 
+async function appReady(): Promise<boolean> {
```

Comment:
I am not sure if this is related but I do see a few seconds delay between when the window shows up and when it starts populating. The fact that splash screen animation is disabled exacerbates the lag feeling. Something to think about in the future releases.

Reply:
it is probably the environment discovery that is taking the time. launching without welcome page and opening a file at launch is heavy on the system. launching a server and discovering environments etc. this is something we can optimize I think. we can take a look in future releases.